uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementLog.php

```diff
@@ -0,0 +1,90 @@
+<?php
+use CRM_HRLeaveAndAbsences_Exception_InvalidLeavePeriodEntitlementLogException as InvalidLeavePeriodEntitlementLogException;
+
+class CRM_HRLeaveAndAbsences_BAO_LeavePeriodEntitlementLog extends CRM_HRLeaveAndAbsences_DAO_LeavePeriodEntitlementLog {
+
+  /**
+   * Create a new LeavePeriodEntitlementLog based on array-data
+   *
+   * @param array $params
+   *
+   * @return CRM_HRLeaveAndAbsences_DAO_LeavePeriodEntitlementLog|NULL
+   */
+  public static function create($params) {
+    $entityName = 'LeavePeriodEntitlementLog';
+    $hook = empty($params['id']) ? 'create' : 'edit';
+
```

Comment:
I think it would be better to prevent the update here instead using the `validateUpdatesNotAllowed` method. One of the reasons is that, if the update won't really happen, there's reason to trigger the pre hook. If we throw an exception right here, the hook will never be triggered.